api/admin/controller/__init__.py

```diff
@@ -1462,9 +1476,10 @@ def _set_integration_library(self, integration, library_info, protocol):
     def _set_integration_settings_and_libraries(self, integration, protocol):
         settings = protocol.get("settings")
         for setting in settings:
-            result = self._set_integration_setting(integration, setting)
-            if isinstance(result, ProblemDetail):
-                return result
+            if setting.get('key') != 'mirror_integration_id':
```

Comment:
In other places (e.g. line 1398) it's possible for there to be multiple integrations of this type, all ending with "mirror_integration_id", but here, no value other than "mirror_integration_id" will work. I suspect the other code is right and this needs to be changed to use `endswith`.